src/modules/editor/SpriteEditor/SpriteEditor.tsx

```diff
@@ -1,28 +1,228 @@
+
+      setDragStart({ x: e.clientX, y: e.clientY });
+    } else if (isDrawing) {
+      const rect = e.currentTarget.getBoundingClientRect();
+      const spriteX = Math.floor(((e.clientX - rect.left) / rect.width * zoom) - (Math.floor(position.x) / 8));
+      const spriteY = Math.floor(((e.clientY - rect.top) / rect.height * zoom) - (Math.floor(position.y) / 8));
+      const x = Math.floor((e.clientX - rect.left) / (rect.width / zoom) * 8 - Math.floor(position.x)) % 8;
+      const y = Math.floor((e.clientY - rect.top) / (rect.width / zoom) * 8 - Math.floor(position.y)) % 8;
+      const spriteIndex = x + spriteX * 8;
+      const pixelIndex = y + spriteY * 8;
+      handleClick(spriteIndex, pixelIndex);
+    }
+  };
+
+  const handleMouseUp = (e: React.MouseEvent<HTMLCanvasElement>): void => {
+    if (e.button === 2) {
+      setIsDragging(false);
+    } else if (e.button === 0) {
+      setIsDrawing(false);
+    }
+  };
+
+  return (
+    <div className="editor-layout">
+      <div className="canvas-container">
+        <div className="sprite-editor-header">
+          <div className="color-selector">
+            {colorPalette01.map((color, index) => (
+              <button
+                key={color.name}
+                onClick={() => changeColor(index)}
+                style={{ backgroundColor: color.hex }}
+                className={`color-button ${currentColor === index ? "selected" : ""}`}
+                title={color.name}
+              />
+            ))}
+          </div>
+          <div ref={canvasContainerRef} className="draw-canvas-container">
+            <StyledCanvas
+              ref={drawCanvasRef}
+              spriteSheet={canvasSpriteSheet}
+              screenSize={DrawCanvasSize}
+              palette={palette}
+              onWheel={handleWheel}
+              onMouseDown={handleMouseDown}
+              onMouseMove={handleMouseMove}
+              onMouseUp={handleMouseUp}
+              onMouseEnter={() => setIsMouseOverCanvas(true)}
+              onMouseLeave={() => {
+                setIsMouseOverCanvas(false);
+                setIsDragging(false);
+                setIsDrawing(false);
+              }}
+              onClick={(e: React.MouseEvent<HTMLCanvasElement>) => {
+                if (!isDragging && !isDrawing) {
+                  const rect = e.currentTarget.getBoundingClientRect();
+                  const { spriteX, spriteY } = getSpritePos(e, rect, zoom, position);
+                  const { x, y } = getPixelPos(e, rect, zoom, position);
+                  const spriteIndex = x + spriteX * 8;
+                  const pixelIndex = y + spriteY * 8;
+                  handleClick(spriteIndex, pixelIndex);
+                }
+              }}
+            />
+          </div>
+        </div>
+      </div>
+    </div>
+  );
+};
+
+export const spriteEditorTabData = {
+  title: "Sprite",
+  icon: "sprite",
+};
+
+function getPixelPos(e: React.MouseEvent<HTMLCanvasElement, MouseEvent>,
```

Comment:
I believe these two functions should be moved before the declaration of the React component.